Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM docker.io/library/alpine:3.15 as runtime
+
+# TODO: Adjust binary file name
+ENTRYPOINT ["go-bootstrap"]
+
+RUN \
+  apk add --update --no-cache \
+    bash \
+    curl \
+    ca-certificates \
+    tzdata
+
+COPY go-bootstrap /usr/bin/
+USER 65536:0
```

Comment:
Any reason for these particular UID and GID?

Reply:
To make it run as non-root by default and be OpenShift compatible.
Basically, if the image runs as artibrary UID with GID=0 then OpenShift compatibility is given in 99% of the cases (e.g. file system permissions).

Reply:
I never tried to run something as GID 0, but that makes sense.

Reply:
It's required by OpenShift: https://docs.openshift.com/container-platform/4.9/openshift_images/create-images.html#images-create-guide-openshift_create-images

> By default, OpenShift Container Platform runs containers using an arbitrarily assigned user ID. [...]
For an image to support running as an arbitrary user, directories and files that are written to by processes in the image must be owned by the root group and be read/writable by that group. Files to be executed must also have group execute permissions.